### PR TITLE
MFE role: fix branding override

### PR DESCRIPTION
**Pull request by morenol**
Part 2 of the MFE deployment reference for the community.

In this PR, it was fixed the part that install the npm aliases for the MFEs.

Specifically, I used:

https://github.com/eduNEXT/frontend-component-header-edunext/ by changing the value of  MFE_NPM_OVERRIDES to
```yaml
MFE_NPM_OVERRIDES: [
   "@edx/frontend-component-header@git+https://github.com/edunext/frontend-component-header-edunext.git"
]
```

So right now, https://profile.mfe.stage.edunext.co/ and https://account.mfe.stage.edunext.co/ are using the header of that repo (I only changed the logo and some extra changes to the package.json in order to allow the installation from github).

I also started to add documentation for the role.

Configuration Pull Request
---

Make sure that the following steps are done before merging:

  - [ ] A DevOps team member has approved the PR if it is code shared across multiple services and you don't own all of the services.
  - [ ] Are you adding any new default values that need to be overridden when this change goes live? If so:
    - [ ] Update the appropriate internal repo (be sure to update for all our environments)
    - [ ] If you are updating a secure value rather than an internal one, file a DEVOPS ticket with details.
    - [ ] Add an entry to the CHANGELOG.
  - [ ] If you are making a complicated change, have you performed the proper testing specified on the [Ops Ansible Testing Checklist](https://openedx.atlassian.net/wiki/display/EdxOps/Ops+Ansible+Testing+Checklist)?  Adding a new variable does not require the full list (although testing on a sandbox is a great idea to ensure it links with your downstream code changes).
  - [ ] Think about how this change will affect Open edX operators.  Have you updated the wiki page for the next Open edX release?

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @morenol! I've created [OSPR-5015](https://openedx.atlassian.net/browse/OSPR-5015) to keep track of it in JIRA, where we prioritize reviews. Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.

**Comment by morenol** [comment]
Hi @regisb, @giovannicimolin, this is a continuation to the PR that was merged today. Where do you think that I should put the documentation and the indications to use this role?

**Comment by regisb** [comment]
> Where do you think that I should put the documentation and the indications to use this role?

@morenol I think it's best if you create an Architecture Decision Record (ADR). It's a big scary name for a simple rst file that will reside in the documentation/ folder and which will explain how to use the MFE role, why it was decided to implement them this way, and who gets to maintain them (aka: not edX). But maybe @nedbat has a different opinion?

**Comment by nedbat** [comment]
An [ADR](https://open-edx-proposals.readthedocs.io/en/latest/oep-0019-bp-developer-documentation.html#adrs) in this repo is a fine place to start.  We can put the information somewhere else later also.

But before writing the ADR, maybe we should settle the [subdomain vs path issue](https://openedx.atlassian.net/browse/BTR-14).

**Comment by natabene** [comment]
@morenol Thank you for your contribution.

**Comment by xitij2000** [comment]
@morenol  One thing I would love to see is a role to deploy custom/in-progress MFEs. The existing `mfe` can easily do this if the MFE doesn't use any custom build steps, all that's needed is a way to provide data about these custom MFEs and a role to deploy them. 

For instance, if I wanted to deploy a custom MFE  I would be able to do it as:

```yaml
CUSTOM_MFES:
    - repo: http://path.to/repo.git
      name: custom-name
      overrides: []
      additional_settings:
        CUSTOM_SETTING: ""
```
A new role can iterate over this and set up all the MFEs there. 
I can make a PR for this if needed.

**Comment by morenol** [comment]
@xitij2000 I think that it would be a great idea, specially if we want to follow the approach mentioned by @giovannicimolin in https://openedx.atlassian.net/browse/BTR-14?focusedCommentId=495166, where we can use all these CUSTOM_MFES to generate a single nginx file.

I particullarly tried to do that, but I had problems with the interpolation of variables when using the include_role module of ansible. i.e.

I tried to do something like:

```yaml

- name: Build MFE
  include_role:
     name: mfe
  vars: 
    MFE_NAME: "{{ custom_mfe.name }}"
  loop: "{{ CUSTOM_MFES }}"
  loop_control:
    loop_var: custom_mfe
```

And it worked fine, but then I failed to add a way to override all the possible configurations used in MFE from that role, i.e.

```yaml

- name: Build MFE
  include_role:
     name: mfe
  vars: 
    MFE_NAME: "{{ custom_mfe.name }}"
    MFE_REPO: "{{ custom_mfe.repo | default('frontend-app-{{ custom_mfe.name }}') }}"
  loop: "{{ CUSTOM_MFES }}"
  loop_control:
    loop_var: custom_mfe
```
Failed for me when running the npm install task (and not in the git clone task). If you know how to solve that, or if you have another approach to solve that, I think that it would be great to have that

**Comment by xitij2000** [comment]
> And it worked fine, but then I failed to add a way to override all the possible configurations used in MFE from that role, i.e.

That is an interesting issue! I assumed something like this would work:
```yaml
- name: Build MFE
  include_role:
     name: mfe
  vars:  "{{ custom_mfe }}"
  loop: "{{ CUSTOM_MFES }}"
  loop_control:
    loop_var: custom_mfe
```

But it seems that is not supported by ansible. One approach is to make the MFE role explicitly aware of this, i.e. pass it input from the loop directly, and let it build the other variables based on that. I can't think of any other way that would work unfortunately.

**Comment by felipemontoya** [comment]
I did a complete test with all the necessary PRs using the native install and left all the instructions in https://discuss.openedx.org/t/deploying-mfes-in-the-community/2868/22.

If this test is acceptable we can approve and merge this so that it makes the cut of Koa next monday (Nov 9).

**Comment by felipemontoya** [comment]
@morenol this probably needs a rebase

File: playbooks/roles/mfe/tasks/main.yml [hunk 1]
```diff
@@ -69,19 +81,18 @@
   become_user: "{{ MFE_USER }}"
   environment: "{{ MFE_ENVIRONMENT }}"
   tags:
-    - install
-    - install:app-requirements
+    - install:base
```
Comment (davidjoy) [inline playbooks/roles/mfe/tasks/main.yml]: Not sure I follow this change, and the similar one below it in 'install npm overrides'.  

Naive question, but what's the difference between tags of `install` and `install:app-requirements` vs just a tag of `install:base`?  And why are we replacing both with `install:base` here, but below we only replace `install:app-requirements` with `install:base`?

Comment (morenol) [reply playbooks/roles/mfe/tasks/main.yml]: I will remove these tag changes, I only made them in order to test this role with the tools that are in the docker folder of this repo.

**Review by davidjoy — APPROVED** [review]

**Comment by openedx-webhooks** [comment]
@morenol 🎉 Your pull request was merged!

Please take a moment to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=edx/configuration&entry.867055334=https://github.com/edx/configuration/pull/6049&entry.1484655318=https://github.com/morenol&entry.752974735=2020-09-28+22:38&entry.1917517419=2020-11-10+22:36&entry.2133058324=Yes) so we can improve your experience in the future.

**Pull request merged**